src/object/srv_ec.c

```diff
@@ -138,7 +140,8 @@ obj_ec_rw_req_split(daos_unit_oid_t oid, struct obj_iod_array *iod_array,
 	} else {
 		D_ASSERT(leader_id == PO_COMP_ID_ALL);
 
-		leader = oid.id_shard - start_shard;
+		D_ASSERT(oca != NULL);
+		leader = oid.id_shard % obj_ec_tgt_nr(oca);
```

Comment:
just confirm seems should be same behavior as original code? or any case may cause difference?

Reply:
start_shard is the shard_index in the object layout.  but id_shard is real shard id, which are different during reintegration, since reintegrating shard is being added to the object layout.

so it has to use id_shard % obj_ec_tgt_nr() to find out which is the leader.